ecl/hthor/hthorkey.cpp

```diff
@@ -744,11 +744,12 @@ void CHThorIndexReadActivityBase::verifyIndex(IKeyIndex * idx)
     keySize = idx->keySize();
     if (eclKeySize.isFixedSize())
     {
+        unsigned fileposSize = idx->hasSpecialFileposition() ? sizeof(offset_t) : 0;
         if(layoutTrans)
             layoutTrans->checkSizes(df->queryLogicalName(), eclKeySize.getFixedSize(), keySize);
         else
-            if (keySize != eclKeySize.getFixedSize())
-                throw MakeStringException(0, "Key size mismatch reading index %s: index indicates size %u, ECL indicates size %u", df->queryLogicalName(), keySize, eclKeySize.getFixedSize());
+            if (keySize != eclKeySize.getFixedSize() + fileposSize)
```

Comment:
Is this correct?

Reply:
Yes - for backwards compatibility the eclKeySize does not include the fileposition field, but keySize has now been increased to include the size - since it is appended to the buffer.